contracts/MerkleLib.sol

```diff
@@ -2,65 +2,22 @@
-        // The following arrays are required to be the same length. They are parallel arrays for the given chainId and should be ordered by the `l1Tokens` field.
-        // All whitelisted tokens with nonzero relays on this chain in this bundle in the order of whitelisting.
-        address[] l1Tokens;
-        uint256[] bundleLpFees; // Total LP fee amount per token in this bundle, encompassing all associated bundled relays.
-        // This array is grouped with the two above, and it represents the amount to send or request back from the
-        // SpokePool. If positive, the pool will pay the SpokePool. If negative the SpokePool will pay the HubPool.
-        // There can be arbitrarily complex rebalancing rules defined offchain. This number is only nonzero
-        // when the rules indicate that a rebalancing action should occur. When a rebalance does not occur,
-        // runningBalances for this token should change by the total relays - deposits in this bundle. When a rebalance
-        // does occur, runningBalances should be set to zero for this token and the netSendAmounts should be set to the
-        // previous runningBalances + relays - deposits in this bundle.
-        int256[] netSendAmounts;
-        // This is only here to be emitted in an event to track a running unpaid balance between the L2 pool and the L1 pool.
-        // A positive number indicates that the HubPool owes the SpokePool funds. A negative number indicates that the
-
-        // SpokePool owes the HubPool funds. See the comment above for the dynamics of this and netSendAmounts
-        int256[] runningBalances;
-    }
-
-    // This leaf is meant to be decoded in the SpokePool in order to pay out individual relayers for this bundle.
-    struct DestinationDistribution {
-        // Used as the index in the bitmap to track whether this leaf has been executed or not.
-        uint256 leafId;
-        // Used to verify that this is being decoded on the correct chainId.
-        uint256 chainId;
-        // This is the amount to return to the HubPool. This occurs when there is a PoolRebalance netSendAmount that is
-        // negative. This is just that value inverted.
-        uint256 amountToReturn;
-        // The associated L2TokenAddress that these claims apply to.
-        address l2TokenAddress;
-        // These two arrays must be the same length and are parallel arrays. They should be order by refundAddresses.
-        // This array designates each address that must be refunded.
-        address[] refundAddresses;
-        // This array designates how much each of those addresses should be refunded.
-        uint256[] refundAmounts;
-    }
-
     /**
      * @notice Verifies that a repayment is contained within a merkle root.
      * @param root the merkle root.
      * @param rebalance the rebalance struct.
      * @param proof the merkle proof.
      */
-    function verifyPoolRebalance(
+    function verifyPoolRebalanceLeaf(
```

Comment:
why rename this function to end with "Leaf"? I think its not neccessary since you take in a param called "leaf", but if you do go this route then I think you should end `verifyRelayData` and `verifyRelayerDistribution` also with "Leaf"

Reply:
Perhaps `verifyRelayData` is more appropriately named `verifyRelayDataHashLeaf` or `verifyRelayHashLeaf`

Reply:
Okay, fixed on both! I reverted the Leaf change here (it was a find-and-replace issue I think)